networking_aci/plugins/ml2/drivers/mech_aci/common.py

```diff
@@ -72,3 +74,26 @@ def get_address_scope_name(self, context, subnet_pool_id):
 
 def get_segments(context, network_id):
     return ml2_db.get_network_segments(context, network_id)
+
+
+def get_switch_from_local_link(binding_profile):
+    if binding_profile:
+        try:
+            if not isinstance(binding_profile, dict):
+                binding_profile = ast.literal_eval(binding_profile)
+
+            lli = binding_profile.get('local_link_information')
+            # TODO validate assumption that we have 1 lli in list.
+            if lli and lli[0] and isinstance(lli[0], dict):
+                switch = lli[0].get('switch_info', None) or lli[0].get('switch_id', None)
+                if switch:
+                    LOG.info("Using link local information for binding host %s", switch)
+                    return switch
+                else:
+                    LOG.error("Cannot determine switch for local link info %s in binding profile %s.",
+                              lli[0], binding_profile)
+            else:
+                LOG.error("Local information %s is invalid in binding profile %s.",
+                          lli, binding_profile)
+        except ValueError:
+            LOG.info("binding Profile %s cannot be parsed", binding_profile)
```

Comment:
Would it make sense to also move these lines up to the `ast.literal_eval`? And also move the `try` into the `if`? That's what it's there for, right?

Reply:
The method is only copied from `driver.py` over to this file, I didn't change much in it. If we were to improve it what do you think about tossing out `ast.literal_eval`, use `json.loads` and catch that value error directly there?

Reply:
I'd prefer `json.loads` and yes, catch it where it's raised.